sdk/core/Azure.Core.TestFramework/src/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -12,94 +14,198 @@ namespace Azure.Core.TestFramework
 {
     public class DiagnosticScopeValidatingInterceptor : IInterceptor
     {
+        private static readonly MethodInfo ValidateDiagnosticScopeMethod = typeof(DiagnosticScopeValidatingInterceptor).GetMethod(nameof(AwaitAndValidateDiagnosticScope), BindingFlags.NonPublic | BindingFlags.Static);
         public void Intercept(IInvocation invocation)
         {
             var methodName = invocation.Method.Name;
-            if (methodName.EndsWith("Async"))
-            {
-                Type declaringType = invocation.Method.DeclaringType;
-                var ns = declaringType.Namespace;
-                var expectedName = declaringType.Name + "." + methodName.Substring(0, methodName.Length - 5);
-                using ClientDiagnosticListener diagnosticListener = new ClientDiagnosticListener(s => s.StartsWith("Azure."), asyncLocal: true);
-                invocation.Proceed();
 
-                bool expectFailure = false;
-                bool skipChecks = false;
+            Type declaringType = invocation.Method.DeclaringType;
+            var expectedName = declaringType.Name + "." + methodName.Substring(0, methodName.Length - 5);
+            bool strict = !invocation.Method.GetCustomAttributes(true).Any(a => a.GetType().FullName == "Azure.Core.ForwardsClientCallsAttribute");
 
-                bool strict = !invocation.Method.GetCustomAttributes(true).Any(a => a.GetType().FullName == "Azure.Core.ForwardsClientCallsAttribute");
-                if (invocation.Method.ReturnType.Name.Contains("Pageable") ||
-                    invocation.Method.ReturnType.Name.Contains("IAsyncEnumerable"))
+            if (invocation.Method.ReturnType is {IsGenericType: true} genericType &&
```

Comment:
Why don't we add the validator for Pageable in addition to AsyncPageable?

Reply:
This validator runs before we forward calls from async->sync so everything would look sync to it.